concourse/steps/version.mako

```diff
@@ -21,13 +22,22 @@ path_to_repo_version_file = os.path.join(
 output_version_file = os.path.join(job_step.output('version_path'), 'version')
 legacy_version_file = os.path.join(job_step.output('version_path'), 'number')
 
-if (read_callback := version_trait.read_callback()):
+# Assign empty string to calbacks if None, as we'd template 'None' (the string) later otherwise
+if (read_callback := version_trait.read_callback() or ''):
   read_callback = os.path.join(main_repo.resource_name(), read_callback)
 
-# Assign empty string if None, as we'd template 'None' (the string) later otherwise
+
 if (write_callback := version_trait.write_callback() or ''):
   write_callback = os.path.join(main_repo.resource_name(), write_callback)
 
+# Validate that either both callbacks are set or none
+if (write_callback == '' and read_callback != '') or (write_callback != '' and read_callback == ''):
```

Comment:
I think it would be more "pythonic" to check this leveraging coercion to bool, like so:
`if write_callback and (not read_callback) or (not write_callback) and read_callback`

You might also use XOR (not specifically pythonic), like so (I personally prefer this style as it is shorter/more expressive):
`if bool(write_callback) ^ bool(read_callback)`

On an additional note: how about moving this check to `concourse.model.traits.version.py` (define a new method `validate` - this is automatically called if it exists. [example](https://github.com/gardener/cc-utils/blob/master/concourse/model/traits/component_descriptor.py#L435-L447)?

Reply:
done. I decided against the XOR (unless you really insist on it) as, at least for me, it is less readable. I rarely encounter this in any code, so I'm trying to avoid it, if possible.